src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +35,11 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.iterate(0, i -> i + 1)
+                .limit(numbers.size())
+                .map(i -> i % 2 == 0 ? numbers.get(i) : numbers.get(i) - 1)
+                .filter(i -> i % 2 != 0)
+                .average().orElseThrow(() -> new NoSuchElementException());
```

Comment:
```suggestion
                .average()
                .get();
```
можна просто викликати get() якщо Optional пустий, то за замовчуванням викенеться NoSuchElementException